### PR TITLE
Issue #3517: Add a search relevancy factor for content title match.

### PR DESCRIPTION
**Pull request by jenlampton**
Fixes https://github.com/backdrop/backdrop-issues/issues/3517

### CODE REVIEW

File: core/modules/node/node.module [hunk 1]
```diff
@@ -1470,7 +1470,7 @@ function node_search_execute($keys = NULL, $conditions = NULL) {
 /**
  * Implements hook_ranking().
  */
-function node_ranking() {
+function node_ranking($keywords = NULL) {
```
Comment (docwilmot) [inline core/modules/node/node.module:1473]: How does `$keywords` get passed to this? `module_invoke_all()` doesnt do that.

File: core/modules/node/node.module [hunk 2]
```diff
@@ -1322,7 +1322,7 @@ function node_views_api() {
  * @param SearchQuery|SelectQueryExtender $query
  *   A query object that has been extended with the Search DB Extender.
  */
-function _node_rankings(SelectQueryExtender $query) {
+function _node_rankings(SelectQueryExtender $query, $keywords = NULL) {
   if ($ranking = module_invoke_all('ranking')) {
```
Comment (docwilmot) [inline core/modules/node/node.module]: Would need to pass `$keywords` here, or use `module_implements()`

File: core/modules/node/node.module (hunk 1 above)
Comment (jenlampton) [reply core/modules/node/node.module:1473]: they are passed as the 3rd parameter in `module_invoke_all()`. see https://api.backdropcms.org/api/backdrop/core%21includes%21module.inc/function/module_invoke_all/1

File: core/modules/node/node.module (hunk 2 above)
Comment (jenlampton) [reply core/modules/node/node.module]: yes, see line 1434

Comment (jenlampton) [reply core/modules/node/node.module]: Ah, right. Missed this one :)

File: core/modules/node/node.module [hunk 3]
```diff
@@ -1488,6 +1488,12 @@ function node_ranking() {
       // The promote flag is either 0 or 1, which is automatically normalized.
       'score' => 'n.promote',
     ),
+    'title_match' => array(
+      'title' => t('Node title match'),
+      // If keys appears in the title, return 1, else 0.
```
Comment (klonos) [inline core/modules/node/node.module]: Either change `keys` to singular, or change the verb to `appear`.

File: core/modules/node/node.module [hunk 4]
```diff
@@ -1488,6 +1488,12 @@ function node_ranking() {
       // The promote flag is either 0 or 1, which is automatically normalized.
       'score' => 'n.promote',
     ),
+    'title_match' => array(
+      'title' => t('Node title match'),
```
Comment (klonos) [inline core/modules/node/node.module]:
```suggestion
      'title' => t('Content title match'),
```

"Node" is a no-no for the UI, remember? 😉

File: core/modules/node/node.module (hunk 3 above)
Comment (jenlampton) [reply core/modules/node/node.module]: okay.

File: core/modules/node/node.module [hunk 5]
```diff
@@ -1488,6 +1490,12 @@ function node_ranking() {
       // The promote flag is either 0 or 1, which is automatically normalized.
       'score' => 'n.promote',
     ),
+    'title_match' => array(
+      'title' => t('Content title match'),
+      // If the keywords appear in the title, return 1, else 0.
+      'score' => "IF(CONCAT(',', title, ',') REGEXP ',(:words),') > 0, '1','0')",
```
Comment (jenlampton) [inline core/modules/node/node.module]: That's strange. This works locally, but not on the sandbox server. I wonder why? Both are running PHP 5.3. My local is running `MySQL, MariaDB, or equivalent version 5.6.35` and the sandbox is running `MySQL, MariaDB, or equivalent version 5.5.48-37.8`

File: core/modules/node/node.module [hunk 6]
```diff
@@ -1488,6 +1490,12 @@ function node_ranking() {
       // The promote flag is either 0 or 1, which is automatically normalized.
       'score' => 'n.promote',
     ),
+    'title_match' => array(
+      'title' => t('Content title match'),
+      // If the keywords appear in the title, return 1, else 0.
+      'score' => "IF((CONCAT(',', title, ',') REGEXP ',(:words),' > 0), '1','0')",
```
Comment (docwilmot) [inline core/modules/node/node.module:1496]: The code is Greek to me, but shouldn't this be n.title?

**Comment by jenlampton** [comment]
closing for new sandbox

File: core/modules/node/node.module (hunk 6 above)
Comment (jenlampton) [reply core/modules/node/node.module:1496]: Maybe! I'll try that.

**Comment by jenlampton** [comment]
sandbox me.

**Comment by jenlampton** [comment]
Bah...  creating new PR.

**Comment by jenlampton** [comment]
replaced by https://github.com/backdrop/backdrop/pull/2499

**Pull request closed without merging**